### DISCUSSION TITLE
Dotenv - after migration 5.4 -> 6.0 new signature

### DISCUSSION THREAD

**Discussion by Quovandius** [symfony/symfony#46127] [Q&A] [answered]
### Symfony version(s) affected

6.0

### Description

Hello everyone,

I have prepared a migration to the new major version 5.4.4 -> 6.0.7
I have some issue to fix on my current project, that's obvious ^^

But i have fix a fatal error for dotenv component. The error message is :
Fatal error: Uncaught TypeError: Symfony\Component\Dotenv\Dotenv::__construct(): Argument #1 ($envKey) must be of type string, bool given, called in /srv/app/config/bootstrap.php on line 18 and defined in /srv/app/vendor/symfony/dotenv/Dotenv.php:43 Stack trace: #0 /srv/app/config/bootstrap.php(18): Symfony\Component\Dotenv\Dotenv->__construct(true) #1 /srv/app/public/index.php(7): require('/srv/app/config...') #2 {main} thrown in /srv/app/vendor/symfony/dotenv/Dotenv.php on line 43

To fix this error message, i have update the file : config/bootstrap.php on line 19 with this config :
```
    // load all the .env files
    (new Dotenv('true'))->loadEnv(dirname(__DIR__) . '/.env');
```

The parameter must be a string and not a boolean anymore.

I don't know if its an obvious issue, or its a bug due to upgrade major version symfony. I have made some research, and I don't find any topic about this.

Thanks for your help.

### How to reproduce

To reproduce it on my project, when I have pass successfully the migration on symfony 6, we just need to update the bootstrap.php file and update this line like this :

`(new Dotenv(true))->loadEnv(dirname(__DIR__) . '/.env');`

### Possible Solution

We just need to update this param, and that's work.

### Additional Context

This file is generated by symfony project, but a this time, this isn't a vendor file - so I don't know if the process of migration can be update this file, or if its the duty of our team.

**Comment by stof** [comment, marked as answer]
The 5.4 branch of the component is triggering a deprecation for that case: https://github.com/symfony/symfony/blob/7f37797e0a352798a28cdb9b98193909dced700e/src/Symfony/Component/Dotenv/Dotenv.php#L49

However, this deprecation might happen too early in the bootstrapping process to be caught by the tooling reporting them.

> This file is generated by symfony project, but a this time, this isn't a vendor file - so I don't know if the process of migration can be update this file, or if its the duty of our team.

Flex has a mechanism to update projects to use the latest version of recipes. Using it (even before doing the Symfony 6 migration, as the Symfony 5.4 recipe does not rely on the deprecated signature) would have updated that file.

**Comment by Quovandius** [comment]
Thanx for this quick answer. So could I purpose a fix on flex to upgrade this file with symfony 5.4 ? if yes, that's my first contribution ^^

**Comment by stof** [comment]
@Quovandius the recipe for Symfony 5.1 already removed usage of the deprecated argument: https://github.com/symfony/recipes/blob/master/symfony/framework-bundle/5.1/public/index.php#L10
and the recipes for Symfony 5.3+ are relying on the symfony/runtime component, which moves that logic to the vendor folder.

So there is nothing to contribute to Symfony. There is something to contribute to your own project.

**Comment by Quovandius** [comment]
Ok, I can close this issue. Thanks for your answer.
